flagd/pkg/service/flag-sync/sync_service.go

```diff
@@ -39,14 +43,43 @@ type Service struct {
 	startupTracker syncTracker
 }
 
+func loadTLSCredentials(certPath string, keyPath string) (credentials.TransportCredentials, error) {
+	// Load server's certificate and private key
+	serverCert, err := tls.LoadX509KeyPair(certPath, keyPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load key pair from certificate paths '%s' and '%s': %w", certPath, keyPath, err)
+	}
+
+	// Create the credentials and return it
+	config := &tls.Config{
+		Certificates: []tls.Certificate{serverCert},
+		ClientAuth:   tls.NoClientCert,
+		MinVersion:   tls.VersionTLS12,
+	}
+
+	return credentials.NewTLS(config), nil
+}
+
 func NewSyncService(cfg SvcConfigurations) (*Service, error) {
 	l := cfg.Logger
 	mux, err := NewMux(cfg.Store, cfg.Sources)
 	if err != nil {
 		return nil, fmt.Errorf("error initializing multiplexer: %w", err)
 	}
 
-	server := grpc.NewServer()
+	var server *grpc.Server
+	if cfg.CertPath != "" && cfg.KeyPath != "" {
+		tlsCredentials, err := loadTLSCredentials(cfg.CertPath, cfg.KeyPath)
+		if err != nil {
+			l.warn("Couldn't load credentials from given paths. Starting server without TLS connection...", err)
```

Comment:
Is continuing to start-up the right decision here? I think perhaps it makes for the process to exit at this point or else you're potentially running a less secure app than you think you are.

Reply:
yes you're right, it make sense to "throw` here, instead of continuing, but with a proper error message.

Reply:
I've added `return nil, fmt.Errorf("failed to load TLS cert and key: %w", err)` (similar to other TLS setups, and a test case related to this as well.